src/lib/kernel/LinuxProcess.cc

```diff
@@ -76,7 +83,7 @@ LinuxProcess::LinuxProcess(span<char> instructions, YAML::Node config)
 
 LinuxProcess::~LinuxProcess() {
   if (isValid_) {
-    delete[] processImage_;
+    free(processImage_);
```

Comment:
Have we considered a smart pointer approach with `std::shared_ptr<char*> processImage_`

Reply:
I'd argue it's a desirable approach for a C++17 project which already makes heavy use of them

Reply:
PTAL